### PR TITLE
Allow users to filter the hits to use

### PR DESCRIPTION
**Pull request by mikaa123**
The `hitToParams` function can now return a falsy value.
To achieve this, this commit chunks results and only flushes when it hits 50k.

### CODE REVIEW

**Comment by Haroenv** [comment]
The batching and flushing seems nice too, but could you do the two things in separate PRs?

File: index.js [hunk 1]
```diff
@@ -2,33 +2,60 @@ const algoliasearch = require('algoliasearch');
...
+  };
+
+  const flush = () => {
+    const chunks = [];
+    let chunk = [];
+    batch.forEach(entry => {
+      if (chunk.length < CHUNK_LENGTH) {
+        chunk.push(entry);
+      }
+      if (chunk.length === CHUNK_LENGTH) {
```
Comment (Haroenv) [inline index.js]: What if it's bigger than CHUNK_LENGTH?

File: index.js [hunk 2]
```diff
@@ -2,33 +2,60 @@ const algoliasearch = require('algoliasearch');
 const { createSitemapindex, createSitemap } = require('./sitemap');
 const { saveSiteMap } = require('./saveFiles');
 
+const CHUNK_LENGTH = 50000;
```
Comment (Haroenv) [inline index.js]: CHUNK_SIZE sounds better to me

File: sitemap.js [hunk 3]
```diff
@@ -39,25 +39,23 @@ function createSitemap({ hits, hitToParams }) {
...
+        rel: 'alternate',
+        hreflang: language,
+        href: hitToURL(language)
+      }));
     }
     return sitemap;
   };
 
-  hits.forEach(hit => addURL(hitToParams(hit)));
+  entries.forEach(addURL);
```
Comment (Haroenv) [inline sitemap.js]: how come this isn't used anymore?

File: index.js (hunk 2 above)
Comment (mikaa123) [inline index.js]: true!

File: index.js (hunk 1 above)
Comment (mikaa123) [inline index.js]: It shouldn't be possible we only push a single `entry` at a time in `chunk`

Comment (Haroenv) [inline index.js]: perfect

**Pull request merged**